src/ui.c

```diff
@@ -64,7 +64,7 @@ file_finder_read(struct file_finder *finder, const char *commit)
 			break;
 		}
 
-		finder->file[files] = calloc(1, sizeof(*finder->file) + buf.size);
+		finder->file[files] = calloc(1, sizeof(*finder->file) + buf.size + 1);
```

Comment:
Oh, I thought the first term was just to count the other member of the struct.

The issue is still there as described, because `sizeof(*finder->file)` returns constant `sizeof(uintptr_t)`, coincidentally `size_t`, the size of `matches`, the other member of the struct.

It should be `sizeof(**finder->file)` or `sizeof(struct file_finder_line)`.  The second one is better because it's harder to typo.  And either is better than my `+ 1` because of alignment.